src/card/CreditCard.java

```diff
@@ -67,36 +70,44 @@ public String getCvv() {
 	/**
 	 * Checks if the address associated already exists, if it does then just adds the 
 	 * credit card with a the foreign key to the appropriate existing address. If it can't
-	 * update to the existing address, the card isn't added and method returns false.
+	 * update to the existing address, an exception is thrown.
 	 * If address doesn't exist, adds the address to the database, and then the credit
-	 * card with the appropriate foreign key. Returns false if the new addresses wasn't added.
+	 * card with the appropriate foreign key. If address isn't added, an exception is thrown.
 	 * 
 	 * @param databaseConnection
-	 * @return true if the credit card was successfully added, false if not
+	 * @return true if the credit card was successfully added, false if nickname already in use
+	 * @throws Exception if the 
 	 */
-	public boolean addCard(ConnectionSource databaseConnection) {
+	public boolean addCard(ConnectionSource databaseConnection) throws Exception {
 		if (this.billingAddress.addressExists(databaseConnection)) {
 			boolean updateSuccessful = this.billingAddress.updateToExistingAddress(databaseConnection);
 			this.creditCardEntity.setBillingAddress(this.billingAddress.getAddressEntity());
 			if (!updateSuccessful) {
-				return false;
+				throw(new Exception("Exisitng address not gotten properly, abort add card"));
 			}
 		}
 		else {
 			boolean addAddressSuccessful = this.billingAddress.addAddress(databaseConnection);
 			this.creditCardEntity.setBillingAddress(this.billingAddress.getAddressEntity());
 			if(!addAddressSuccessful) {
-				return false;
+				throw(new Exception("Address not added properly, abort add card"));
 			}
 		}
 		
 		try {
 			Dao<CreditCardEntity, String> creditCardDao = DaoManager.createDao(databaseConnection, CreditCardEntity.class);
+			Map<String, Object> queryParam = new HashMap<String, Object>();
+			queryParam.put("nickname", this.getNickname());
+			List<CreditCardEntity> returnedCreditCards = creditCardDao.queryForFieldValues(queryParam);
```

Comment:
I think you should check the username associated with the credit card as well because two different users can have the same nickname like "BOA"